test/integration/framework/utils.go

```diff
@@ -0,0 +1,307 @@
+							},
+							Env: []corev1.EnvVar{
+								{Name: "TILLER_NAMESPACE", Value: opts.Namespace},
+							},
+							LivenessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/liveness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+							ReadinessProbe: &corev1.Probe{
+								Handler: corev1.Handler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path: "/readiness",
+										Port: intstr.FromInt(44135),
+									},
+								},
+								InitialDelaySeconds: 1,
+								TimeoutSeconds:      1,
+							},
+						},
+					},
+					HostNetwork:  opts.EnableHostNetwork,
+					NodeSelector: nodeSelectors,
+				},
+			},
+		},
+	}
+	return deployment, nil
+}
+
+// getRunningPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getRunningPodByLabels(ctx context.Context, labelsMap map[string]string, seedOrShoot ClusterType) (*corev1.Pod, error) {
+	var (
+		podList *corev1.PodList
+		err     error
+	)
+	set := labels.Set(labelsMap)
+
+	switch seedOrShoot {
+	case Seed:
+		podList, err = o.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	case Shoot:
+		podList, err = o.K8sShootClient.ListPods(metav1.NamespaceSystem, metav1.ListOptions{
+			LabelSelector: set.AsSelector().String(),
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		for _, pod := range podList.Items {
+			o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+			if pod.Status.Phase == corev1.PodRunning {
+				return &pod, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+// getPodByLabels fetches the first running pod with the desired set of labels <labelsMap>
+func (o *GardenerTestOperation) getPodByLabels(ctx context.Context, labelsMap map[string]string, namespace string, client kubernetes.Interface) (*corev1.Pod, error) {
+	var runningPodWithDesiredLabels *corev1.Pod
+	set := labels.Set(labelsMap)
+
+	var podList *corev1.PodList
+	var err error
+
+	podList, err = client.ListPods(namespace, metav1.ListOptions{
+		LabelSelector: set.AsSelector().String(),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Fetch the first running pod of the Kube-apiserver
+	if len(podList.Items) > 0 {
+		pod := podList.Items[0]
+		o.ShootGardenerTest.Logger.Infof("%s:%s", pod.Namespace, pod.Name)
+		runningPodWithDesiredLabels = &pod
+	}
+	return runningPodWithDesiredLabels, nil
+
+}
+
+// GetAdminPassword gets the admin password for authenticating agains the api
+func (o *GardenerTestOperation) getAdminPassword(ctx context.Context) (*string, error) {
+	password, err := getObjectFromSecret(ctx, o.K8sSeedClient, o.Shoot.SeedNamespace, kubecfg, password)
```

Comment:
why not immediately return here?

Reply:
Change the signature of this function to `(string, error)` and do an immediate return.